cmd/boulder/main.go

```diff
@@ -60,57 +56,89 @@ func readAndValidateConfigFile(name, filename string) error {
 	return cmd.ValidateJSONConfig(cv, file)
 }
 
-func main() {
-	cmd.LookupCommand(core.Command())()
+// getConfigPath returns the path to the config file if it was provided as a
+// command line flag. If the flag was not provided, it returns an empty string.
+func getConfigPath() string {
+	for i := 0; i < len(os.Args); i++ {
+		arg := os.Args[i]
+		if arg == "--config" || arg == "-config" {
+			if i+1 < len(os.Args) {
+				return os.Args[i+1]
+			}
+		}
+		if strings.HasPrefix(arg, "--config=") {
+			return strings.TrimPrefix(arg, "--config=")
+		}
+		if strings.HasPrefix(arg, "-config=") {
+			return strings.TrimPrefix(arg, "-config=")
+		}
+	}
+	return ""
 }
 
-func init() {
-	cmd.RegisterCommand("boulder", func() {
+var boulderUsage = fmt.Sprintf(`Usage: %s <subcommand> [flags]
+
+  Each boulder component has its own subcommand. Use --list to see
+  a list of the available components. Use <subcommand> --help to
+  see the usage for a specific component.
+`,
+	core.Command())
+
+func main() {
```

Comment:
The order of the three main stanzas in this function is:
1) Look up the subcommand if the executable name was "boulder"
2) Validate config (which 
3) Look up the subcommand if the executable name was not "boulder"

This feels weird to me. Also, I'm think it's broken (see comment below).

It seems like it should be
1) Figure out the name of the command that is being run
2) Look up the config struct, do config validation
3) Look up the function, execute it